src/sites/dMCDA.jl

```diff
@@ -691,6 +750,80 @@ function vikor(S::Array{Float64,2}; v::Float64=0.5)::Array{Union{Float64,Int64},
+
+Perform site selection for a given domain for multiple scenarios defined in a dataframe.
+    
+# Arguments
+- domain : ADRIA Domain type, indicating geographical domain to perform site selection over.
+- criteria : DataFrame of criteria weightings and thresholds.
+- sumcover : array of size (number of scenarios * number of sites) containing the summed coral cover for each site selection scenario.
+- area_to_seed : area of coral to be seeded at each time step in km^2
+- time_step : time step at which seeding and/or shading is being undertaken.
+
+# Returns
+- ranks_store : number of scenarios * sites * 3 (last dimension indicates: site_id, seeding rank, shading rank)
+    containing ranks for each scenario run.
+"""
+function run_site_selection(domain::Domain, criteria::DataFrame, sumcover::AbstractArray, area_to_seed::Float64, time_step::Int64)
+
+    ranks_store = NamedArray(zeros(size(criteria, 1), domain.sim_constants.nsiteint, 3))
+    idx_rows = ["scen_$i" for i = 1:size(criteria, 1)]
+    setnames!(ranks_store, idx_rows, 1)
+    dhw_scens = domain.dhw_scens
+    wave_scens = domain.wave_scens
+
+    site_data = domain.site_data
+
+    for (cover_ind, scen_criteria) in enumerate(eachrow(criteria))
+
+        max_depth = scen_criteria.depth_min + scen_criteria.depth_offset
+        depth_criteria = (site_data.depth_med .<= max_depth) .& (site_data.depth_med .>= scen_criteria.depth_min)
+        depth_priority = collect(1:size(site_data, 1))[depth_criteria]
+
+        ranks_temp = site_selection(domain, scen_criteria, wave_scens[time_step, :, criteria.wave_scenario[cover_ind]], dhw_scens[time_step, :, criteria.wave_scenario[cover_ind]], depth_priority, sumcover[cover_ind, :, :], area_to_seed)
+        ranks_store[cover_ind, 1:size(ranks_temp, 1), :] = ranks_temp
+    end
+
+    return ranks_store
+end
+
+"""
+    site_selection(domain::Domain, criteria::DataFrameRow{DataFrame,DataFrames.Index}, w_scens::NamedArray, dhw_scens::NamedArray, sumcover::AbstractArray, area_to_seed::Float64)
+
+Perform site selection using a chosen mcda aggregation method, domain, initial cover, criteria weightings and thresholds.
+
+# Arguments
+- criteria : contains criteria weightings and thresholds for a single scenario.
+- mcda_vars : site selection criteria and weightings structure
+- w_scens : array of length nsites containing wave scenario.
+- dhw_scens : array of length nsites containing dhw scenario.
+- sumcover : summed cover (over species) for single scenario being run, for each site.
+- area_to_seed : area of coral to be seeded at each time step in km^2
+
+# Returns
+- ranks: n_reps * sites * 3 (last dimension indicates: site_id, seeding rank, shading rank)
+    containing ranks for single scenario.
+"""
+function site_selection(domain::Domain, criteria::DataFrameRow, w_scens::NamedArray, dhw_scens::NamedArray, site_ids::AbstractArray, sumcover::AbstractArray, area_to_seed::Float64)
```

Comment:
Is there a reason why this function, which appears to also do guided site selections, is named differently?

Reply:
not really, more a legacy from how I first structured the code when making changes...guided_site_selection was called dMCDA() but I changed it to match the function for unguided site selection prior to writing site_selection()... perhaps I should revert the name to dMCDA and change site_selection() to guided_site_selection() ?

Reply:
Leave it as is for now but could you make a note to have a discussion with Ken to nail down a good generic term? ("location"?)

"site" only made sense in the early prototype within-cluster context. But we will be moving to larger spatial scales in the coming weeks/months where deployment is at the reef scale.